### DISCUSSION TITLE
Roadmap for new features

### DISCUSSION THREAD

**Discussion by rdz-oss** [rdz-oss/BattyBirdNET-Pi#26] [Announcements]
Hi everone! Let us talk about new/upcoming  features here!

**Comment by rdz-oss** [comment]
On the Roadmap:

1. Easy install. One image with pre-installed BattyBirdNET-Pi. Use the Raspberry OS Imager, set your WIFI and localization settings, burn and you are ready to go!

Why: Make the system available to more people who do not know or want to deal with the shell, ssh and such.

Status: Prototype working. Coming soon including a new help text - step by step with NO Linux knowledge necessary. Works from Windows, Mac and Linux in just one Step.

2. Add your own classifier via the file menu. Upload your .tflite and labels file via the file manager and it becomes an option in the BattyBirdNET-Pi. Train using the BattyBirdNET-Analyzer.

Why: Can be used with models fine tuned for your region and to detect specific species that are under protection/extinction risk. Make this easy to add in the system so it can be applied in conservation !

Status: Working in my mind, ie I know what to do to get it done. Can be done already by overwriting an existing classifier - works but not elegant. But already an option!

3. Data upload to your personal cloud/NAS. 

Why: Keep your data safe for longer term use. Full control to the user.

Status: Working on prototype for S3 storage.

4. Donate data option for further development of the classifiers. Help to make the system better by donating your data under the ODBL License used also by e.g. OpenStreetMap.

Why: The classifiers provided with battyBirdNET can only be as good as the data used. So please help to make this the best system available. This is possible if you donate data!

Status: Will become available with 3 as an option/choice.

5. Improve the UI performance and appearance options.

Which other features would you like to see? Would you have other priorities ?

Let me know,

Sincereley,
Richard

**Reply by arne1921KF** [reply]
Great idea for a thread!

Maybe tweaking privacy settings should be high on the priority list?

Idea for a workflow:
non-persistent recording (e.g., to ramdisk) and processing from non-persistent memory for distinction of human from non-human parts of the soundscape. *Only* saving to persistent memory (i.e., *any* form of persistent memory) when set privacy threshold is maintained.

**Reply by rdz-oss** [reply]
Hi @arne1921KF ,

good suggestion. It appears that there is a bug in the implementation by BirdNET-Pi. Will put it on the list. Currently human-vocalization can be put on the excluded species list (working) to reduce accidental capture as an additional measure. Also, the audiomoth can be configured to ignore human voice range ( high pass say above 12kHz).

**Comment by simonlinke** [comment]
Sounds great Richard - so I just replace the tflite that exists with my custom classifier? 

Would that actually work with the original distro? (I always wondered that). 

Gruss von einem Allgaeuer in Australien.

**Reply by rdz-oss** [reply]
Hi @simonlinke , I do not think that it will work in the BirdnET-Pi original or other forks as the BirdNET-Analyzer uses an extra layer (classifier) on the output of the trained BirdNET tflite model. That layer is not used in the code of BirdNET-Pi. 
It is used in BattyBirdnET-Pi as that is exactly how it is cross trained for bats!
Fine tuning for birds or crosstraining for bats require an extra classifier which comes after BirdNET. 
Since BattyBirdNET-Pi is also enabled for birds you can use a fine tuned model for birds, too.

I will offer a howto some time. But it is really simple now that I prepared the BattyBirdNET-pi for that.

**Comment by Svardsten53** [comment]
These features would be nice to have (already mentioned by me in the general thread).

1. Adding species to the exclusion list.
3. Activate option to use Flickr images.
